sql/analyzer/resolve_columns.go

```diff
@@ -304,15 +321,44 @@ func qualifyColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel
 			return qualifyExpression(e, n, evalSymbols)
 		})
 		if err != nil {
-			return n, transform.SameTree, err
+			return originalNode, transform.SameTree, err
 		}
-		if checkConstraintsUpdated {
+		if checkConstraintsChanged || groupByChanged {
 			identity = transform.NewTree
 		}
```

Comment:
my mental model for manipulating identity usually uses `sameX`. So introducing the `||` makes sense but might go a bit against the grain of how these are usually used? You'll se a lot of code that looks like `checksSame`, ... `groupSame`, and then `if checksSame && groupSame {}` at the end. I originally wrote it the first way, and then  spent two days rewriting every usage the second way to appease zach.

Reply:
No problemo; I'll refactor to `same && same`. Definitely worth it to be consistent with the other usage. 

It would be neat if there was some way to chain serial transformations together like this so the boilerplate code would be all encapsulated somewhere else.